File: design-patterns/go/composite/transport_list/bike.go
// bike.go

package main

import "fmt"

type Bike struct {
}

func NewBike() (bike *Bike) {
	bike = &Bike{}
	return
}

func (bike *Bike) Start() {
	fmt.Println("Starting Bike...")
}

func (bike *Bike) Stop() {
	fmt.Println("Stopping Bike...")
}

func (bike *Bike) Operate() {
	fmt.Println("Riding Bike")
}

File: design-patterns/ts/composite/transport-list/bike.ts
// bike.ts

import Transport from "./transport";

class Bike implements Transport {
    start(): void {
        console.log("Starting Bike...");
    }

    operate(): void {
        console.log("Riding Bike");
    }

    stop(): void {
        console.log("Stopping Bike...");
    }
}

export default Bike;